pydantic/_internal/_decorators.py

```diff
@@ -489,26 +489,17 @@ def unwrap_wrapped_function(
     Returns:
         The underlying function of the wrapped function.
     """
-    all: list[Any]
+    all: set[Any] = {partial, partialmethod, property}
+
     try:
         from functools import cached_property  # type: ignore
-
-        cached_property_list = [cached_property]
     except ImportError:
-        cached_property = int  # anything that doesn't match isinstance below
-        cached_property_list = []
+        cached_property = type('', (), {})  # type: ignore
```

Comment:
Why is this required?

Reply:
creating a new type to reduce branching later I'm guessing?